header-gossip-network.md

```diff
@@ -0,0 +1,143 @@
+
+The accumulator is defined as an [SSZ](https://ssz.dev/) data structure with the following schema:
+
+```python
+EPOCH_SIZE = 8192 # blocks
+MAX_HISTORICAL_EPOCHS = XXXX
+
+# An individual record for a historical header.
+HeaderRecord = Container[block_hash: bytes32, total_difficulty: uint256]
+
+# The records of the headers from within a single epoch
+EpochAccumulator = List[HeaderRecord, max_length=EPOCH_SIZE]
+
+Accumulator = Container[
+    historical_epochs: List[bytes32, max_length=MAX_HISTORICAL_EPOCHS],
+    current_epoch: EpochAccumulator,
+]
+```
+
+The algorithm for managing the accumulator is as follows.
+
+> TODO: provide a written spec too
+
+```python
+def update_accumulator(accumulator: Accumulator, new_block_header: BlockHeader) -> None:
+    # get the previous total difficulty
+    if len(accumulator.current_epoch) == 0:
+        # genesis
+        last_total_difficulty = 0
+    else:
+        last_total_difficulty = accumulator.current_epoch[-1].total_difficulty
+
+    # check if the epoch accumulator is full.
+    if len(accumulator.current_epoch) == EPOCH_SIZE:
+        # compute the final hash for this epoch
+        epoch_hash = hash_tree_root(accumulator.current_epoch)
+        # append the hash for this epoch to the list of historical epochs
+        accumulator.historical_epochs.append(epoch_hash)
+        # initialize a new empy epoch
+        accumulator.current_epoch = []
+
+    # construct the concise record for the new header and add it to the current epoch.
+    header_record = HeaderRecord(header.hash, last_total_difficulty + header.difficulty)
+    accumulator.current_epoch.append(header_record)
+```
+
+### Growth over time
+
+Each `HeaderRecord` is 64 bytes meaning the `EpochAccumulator` can range from `0 - 64 * EPOCH_SIZE` bytes (0-512KB) over the course of a single epoch.
+
+The `Accumulator.historical_epochs` grows by 32 bytes per epoch.
+
+At a 13 second block time we expect:
+
+- ~1 epochs per day
+- ~6 epochs per week
+- ~25 epochs per month
+- ~296 epochs per year
+
+Ignoring fluxuations in the size of `Accumulator.current_epoch` we should expect the size of the accumulator to grow at a rate of roughly 10kb per year.
+
+
+## Wire Protocol
+
+The Header Gossip Network is an overlay network on the [Discovery V5 Protocol](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-theory.md).  The overlay network uses the PING/PONG/FINDNODES/FOUNDNODES/FINDCONTENT/FOUNDCONTENT/OFFER/ACCEPT messages from the [Portal Wire Protocol](./portal-wire-protocol.md).
+
+### Distance Function
+
+The Header Gossip Network uses the standard XOR distance function.
+
+### PING payload
+
+TODO: specify what a node should do prior to having acquired a copy of the accumulator
```

Comment:
Here are some loose thoughts:

#### Return the "empty" hash

Treat the lack of accumulator as if the node only has the accumulator at the genesis block....

I don't like this option because it can't be disambiguated from the natural case of the chain actually being in the genesis state.

#### Return 0

Use `0x000...000` to signal that a node does not yet have a version of the accumulator.

This seems appropriate because it cleanly signals lack of accumulator.